Source/missiles.cpp

```diff
@@ -590,6 +590,90 @@ void CheckMissileCol(Missile &missile, int mindam, int maxdam, bool shift, Point
 		PlaySfxLoc(MissilesData[missile._mitype].miSFX, missile.position.tile);
 }
 
+/**
+ * @brief Could the missile collide with solid objects? (like walls or closed doors)
+ */
+bool CouldMissileCollideWithSolidObject(Point tile)
+{
+	int oid = dObject[tile.x][tile.y];
+	if (oid != 0) {
+		oid = abs(oid) - 1;
+		if (!Objects[oid]._oMissFlag)
+			return true;
+	}
+	return nMissileTable[dPiece[tile.x][tile.y]];
+}
+
+void MoveMissileAndCheckMissileCol(Missile &missile, int mindam, int maxdam, bool ignoreStart, bool ifCollidesDontMoveToHitTile)
+{
+	Point prevTile = missile.position.tile;
+	missile.position.traveled += missile.position.velocity;
+	UpdateMissilePos(missile);
+	int possibleVisitTiles = (missile.position.velocity.deltaX == 0 || missile.position.velocity.deltaY == 0) ? prevTile.WalkingDistance(missile.position.tile) : prevTile.ManhattanDistance(missile.position.tile);
+
+	int16_t tileTargetHash = dMonster[missile.position.tile.x][missile.position.tile.y] ^ dPlayer[missile.position.tile.x][missile.position.tile.y];

```

Comment:
I prefer lines to be 80, and hard limited to 120 chars. This one is 212 which just barley fits on half of my ultra wide monitor in this differ with 100% zoom.

```suggestion
	Point position = missile.position.tile;
	int possibleVisitTiles = (missile.position.velocity.deltaX == 0 || missile.position.velocity.deltaY == 0) ? prevTile.WalkingDistance(position) : prevTile.ManhattanDistance(position);

	int16_t tileTargetHash = dMonster[position.x][position.y] ^ dPlayer[position.x][position.y];
```
Even with this change it's still 186, but that's a bit better. We can wait with further reducing till it's gets added as a clang rule :)

Reply:
If you want I can change it to this:
```cpp
	int possibleVisitTiles;
	if (missile.position.velocity.deltaX == 0 || missile.position.velocity.deltaY == 0)
		possibleVisitTiles = prevTile.WalkingDistance(missile.position.tile);
	else
		possibleVisitTiles = prevTile.ManhattanDistance(missile.position.tile);
```
This has max 87 chars and avoids a new local variable... but now we get 4 instead of 1 line. 😉

Reply:
funny, I wanted to add manhattan distance recently :D

Reply:
To me long lines are worse then many lines. They are very hard to see structure in and if line length is not standardized eh complexity of two function with the same line count can be vastly different.

I'm not a big fan of uninitialized variables, but seeing this I think it's the best of the 3 options.

Reply:
> funny, I wanted to add manhattan distance recently :D

You are 3 months late :)

Reply:
Changed it. 🙂